Include stress test specific to SSD disks

ajdelgado opened the pull request.

ajdelgado committed:
Add FIO stress test for SSD disks

miloudbagaa commented on files/stress_tests.sh:
```diff
@@ -0,0 +1,281 @@
+}
+
+log_file="${HOME}/${APP_NAME}_$(date +'%Y-%m-%d.%H.%M.%S%z').log"
+vm_method='all'
+cpu_method='all'
+ssd_block_size="4M"
+ssd_workers=64
+ssd_tests_folder="/media/stress_test"
+ssd_volume_group="vg_instances"
+check_ssd=0
```
IMHO, these values should not be hard-coded and passed from maim.yml.

miloudbagaa commented on files/stress_tests.sh:
```diff
@@ -0,0 +1,281 @@
+ssd_volume_group="vg_instances"
+check_ssd=0
+
+need_fio_test && check_ssd=1
```
I guess that only need_fio_test is enough as check_ssd is called in need_fio_test.

ajdelgado replied:
Sorry, this cause confusion. In this line check_ssd is a variable and not a call to the function. I renamed now the function to avoid confusions.

miloudbagaa commented on files/stress_tests.sh:
```diff
@@ -0,0 +1,281 @@
+
+# If all disks are SSD, then run the FIO tests
+if [ "${check_ssd}" == "1" ]; then
+  message "Will do a FIO test in the SSD disks."
```
These messages can be saved in the logging but not in stress-tmp folder.

miloudbagaa replied:
I will explain further the idea behinds my comment. We should have to separate files. The first one for logging and should have only one name without adding the date. You should keep pushing in the log file different events and logs. You can save this log file in /tmp or /var/log. The second file shoud be saved in /root/stress-tmp or $HOME/stress-tmp, where you have to create new files by adding the date to the name of the file stress-test-yyyy-mm-dd-HH-mm.log. In this file, the output should be concise that shows if the stress test has passed or not and also if the FIO test has passed or not.

ajdelgado replied:
I reduced now the output to log file to the minimum needed and the rest will go to syslog (a message written to the log and standard output will also warn that there are more details in syslog).

miloudbagaa commented on files/stress_tests.sh:
```diff
@@ -0,0 +1,281 @@
+  return "${return_code}"
+}
+
+log_file="${HOME}/${APP_NAME}_$(date +'%Y-%m-%d.%H.%M.%S%z').log"
```
Adding the date into log file and without removing it can create affect the HOME folder. An option to save the log files into /tmp.

ajdelgado replied:
Changed

ajdelgado committed:
Change path to log files to tmp folder

ajdelgado committed:
Rename function to avoid confusions

ajdelgado committed:
Reduced log output

matteopozza commented:
A couple of places in which the variable `test_duration` should be probably replaced by `stress_ng_duration`:
- https://github.com/CSCfi/ansible-role-stress/blob/master/tests/test.yml#L6
- https://github.com/CSCfi/ansible-role-stress/blob/master/README.md?plain=1#L20
Readme and test could be also updated to reflect the new FIO addition

ajdelgado committed:
Update README and tests

ajdelgado commented:
> A couple of places in which the variable `test_duration` should be probably replaced by `stress_ng_duration`:
> 
>     * https://github.com/CSCfi/ansible-role-stress/blob/master/tests/test.yml#L6
> 
>     * https://github.com/CSCfi/ansible-role-stress/blob/master/README.md?plain=1#L20
>       Readme and test could be also updated to reflect the new FIO addition

Updated readme and test

ajdelgado committed:
rename test_duration variable
